larq_compute_engine/tflite/tests/bconv2d_test.cc

```diff
@@ -824,16 +743,44 @@ INSTANTIATE_TEST_SUITE_P(
         Values(std::array<int, 2>{1, 1},
                std::array<int, 2>{2, 3}),  // strides height/width
         Values(std::array<int, 2>{1, 1},
-               std::array<int, 2>{3, 2}),  // dilation height/width
-        Values(Padding_VALID, Padding_SAME, Padding_ONE),  // padding
+               std::array<int, 2>{3, 2}),    // dilation height/width
+        Values(Padding_VALID, Padding_ONE),  // padding
         Values(ActivationFunctionType_NONE,
                ActivationFunctionType_RELU),  // activation function
         Values(1, 2),                         // number of threads
         ValuesIn(BConv2DOpTest::GetKernelsTuples(*kKernelMap))),
     TestParam::TestNameSuffix);
 
+// This is the same test as above but now only with zero-padding.
+// Since zero-padding is not supported in combination with ReLu or with the
+// reference implementation (or bitpacked output or int8 output), we only use
+// ActivationFunctionType_NONE and the OPT implementation here,
+// to avoid all the 'SKIPPED' tests.
+INSTANTIATE_TEST_SUITE_P(
+    BigTestSamePadding, BConv2DOpTest,
+    ::testing::Combine(
+        Values(std::array<int, 4>{1, 7, 7, 1}, std::array<int, 4>{1, 8, 5, 1},
+               std::array<int, 4>{1, 7, 7, 64}, std::array<int, 4>{1, 8, 5, 64},
+               std::array<int, 4>{1, 7, 7, 130},
+               std::array<int, 4>{1, 8, 5, 130}),  // input shape [BHWI]
+        Values(std::array<int, 3>{1, 1, 1}, std::array<int, 3>{3, 3, 1},
+               std::array<int, 3>{2, 3, 1}, std::array<int, 3>{1, 1, 4},
+               std::array<int, 3>{3, 3, 4}, std::array<int, 3>{2, 3, 4},
+               std::array<int, 3>{1, 1, 64}, std::array<int, 3>{3, 3, 64},
+               std::array<int, 3>{2, 3, 64}),  // filter shape [HWO]
+        Values(std::array<int, 2>{1, 1},
+               std::array<int, 2>{2, 3}),  // strides height/width
+        Values(std::array<int, 2>{1, 1},
+               std::array<int, 2>{3, 2}),     // dilation height/width
+        Values(Padding_SAME),                 // padding
+        Values(ActivationFunctionType_NONE),  // activation function
+        Values(1, 2),                         // number of threads
+        Values(std::pair<std::string, register_function>{
+            "BConv2DOPT", compute_engine::tflite::Register_BCONV_2D_OPT})),
```

Comment:
Does this mean that we wouldn't test 'same-zero' padding with the reference kernel?

Reply:
The reference kernel doesn't support it (although it should be easy to add), so the that test was already skipped.